src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3444,6 +3478,49 @@ BuildWaitEventSet(List *sessionList)
 }
 
 
+/*
+ * UpdateWaitEventSetFlags modifies the given waitEventSet with the wait flags
+ * for connections in the sessionList.
+ */
+static void
+UpdateWaitEventSetFlags(WaitEventSet *waitEventSet, List *sessionList)
+{
+	int connectionIndex = -1;
```

Comment:
We get some errors `+ERROR:  cannot modify postmaster death event`. I think we should be starting from 1 ` additional 2 is for postmaster and latch` ?

Reply:
> I think we should be starting from 1 additional 2 is for postmaster and latch ?

Oh, no. Those are added to the end of the set.  And, we're not adding the connections that we skip here in `BuildWaitEventSet()`. So, we should not be incrementing `connectionIndex ` when we skip the session in the below loop. Sorry for misguide with my previous comment.